app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,55 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]  
```

Comment:
Unnecessary spacing detected.<br>Use the new Ruby 1.9 hash syntax.<br>Trailing whitespace detected.